x-pack/plugins/rollup/public/crud_app/sections/job_list/detail_panel/detail_panel.test.js

```diff
@@ -0,0 +1,334 @@
+      ({ component } = initTestBed({ isOpen: false }));
+      expect(component.find('DetailPanelUi').children().length).toBeFalsy();
+    });
+
+    it('should show a loading when the job is loading', () => {
+      ({ component, findTestSubject } = initTestBed({ isLoading: true }));
+      const loading = findTestSubject('rollupJobDetailLoading');
+      expect(loading.length).toBeTruthy();
+      expect(loading.text()).toEqual('Loading rollup job...');
+
+      // Make sure the title and the tabs are visible
+      expect(component.find('EuiTab').length).toBeTruthy();
+      expect(component.find('#rollupJobDetailsFlyoutTitle').length).toBeTruthy();
+    });
+
+    it('should display a message when no job is provided', () => {
+      ({ component, findTestSubject } = initTestBed({ job: undefined }));
+      expect(findTestSubject('rollupJobDetailFlyout').find('EuiFlyoutBody').text()).toEqual('Rollup job not found');
+    });
+  });
+
+  describe('tabs', () => {
+    const tabActive = JOB_DETAILS_TAB_SUMMARY;
+    const { component } = initTestBed({ panelType: tabActive });
+    const tabs = component.find('EuiTab');
+    const getTab = (id) => {
+      let selectedTab;
+      tabs.forEach((tab) => {
+        if (tab.text() === tabToHumanizedMap[id].props.defaultMessage) {
+          selectedTab = tab;
+        }
+      });
+      return selectedTab;
+    };
+
+    it('should have 5 tabs visible', () => {
+      const expected = JOB_DETAILS_TABS.map(id => tabToHumanizedMap[id].props.defaultMessage);
+      const tabsLabel = tabs.map(tab => tab.text());
+
+      expect(tabsLabel).toEqual(expected);
+    });
+
+    it('should set default selected tab to the "panelType" prop provided', () => {
+      const tab = getTab(tabActive);
+      expect(tab.props().isSelected).toEqual(true);
+    });
+
+    it('should select the tab when clicking on it', () => {
+      const { job, openDetailPanel } = defaultProps;
+      const termsTab = getTab(JOB_DETAILS_TAB_TERMS);
+
+      expect(termsTab.props().isSelected).toEqual(false);
+
+      termsTab.simulate('click');
+
+      expect(openDetailPanel.mock.calls.length).toBe(1);
+      expect(openDetailPanel.mock.calls[0][0]).toEqual({
+        jobId: job.id,
+        panelType: JOB_DETAILS_TAB_TERMS
+      });
+    });
+  });
+
+  describe('job detail', () => {
+    describe('summary tab content', () => {
+      // Init testBed on the SUMMARY tab
+      const panelType = JOB_DETAILS_TAB_SUMMARY;
+      const { findTestSubject } = initTestBed({ panelType });
+
+      const tabContentSections = ['Logistics', 'DateHistogram', 'Stats']
+        .map((section) => findTestSubject(`rollupJobDetailSummary${section}Section`))
+        .filter(s => !!s.length)
+        .map(section => ({
+          title: section.find('EuiTitle').text(),
+          reactWrapper: section,
+        }));
+
+      it('should have a "Logistics", "Date histogram" and "Stats" section ', () => {
```

Comment:
Looks like this description has an extra space at the end. I think this is probably the most unhelpful comment I've ever left in a code review. 🏆

Reply:
Hahaha 😄